.azuredevops/pipelineTemplates/jobs.validateModuleDeployment.yml

```diff
@@ -165,30 +165,37 @@ jobs:
-                    @{ Name = 'subscriptionId';    Value = '${{ parameters.subscriptionId }}' }
-                    @{ Name = 'managementGroupId'; Value = '${{ parameters.managementGroupId }}' }
-                    @{ Name = "tenantId";          Value = '$(ARM_TENANT_ID)' }
-                    @{ Name = "deploymentSpId";    Value = '$(DEPLOYMENT_SP_ID)' }
-                ) | ForEach-Object { [PSCustomObject]$PSItem }
-
-                # Get additional Custom Parameter File Tokens from input
-                Write-Verbose 'Additional Custom Parameter File Tokens: ${{ deploymentBlock.customParameterFileTokens }}' -Verbose
-                $OtherCustomParameterFileTokens = '${{  deploymentBlock.customParameterFileTokens }}' | ConvertFrom-Json
+                $Settings = Get-Content -Path (Join-Path '$(moduleRepoRoot)' 'settings.json') | ConvertFrom-Json -AsHashTable
 
                 # Construct Token Function Input
                 $ConvertTokensInputs = @{
-                    ParameterFilePath                 = Join-Path '$(parametersRepoRoot)' '${{ deploymentBlock.path }}'
-                    DefaultParameterFileTokens        = $DefaultParameterFileTokens
-                    OtherCustomParameterFileTokens    = $OtherCustomParameterFileTokens
-                    LocalCustomParameterFileTokens    = $Settings.parameterFileTokens.localTokens.tokens
-                    TokenPrefix                       = $Settings.parameterFileTokens.tokenPrefix
-                    TokenSuffix                       = $Settings.parameterFileTokens.tokenSuffix
+                    Tokens = @{
+                        resourceGroupName = '${{ parameters.resourceGroupName }}'
+                        subscriptionId    = '${{ parameters.subscriptionId }}'
+                        managementGroupId = '${{ parameters.managementGroupId }}'
+                        tenantId          = '$(ARM_TENANT_ID)'
+                        deploymentSpId    = '$(DEPLOYMENT_SP_ID)'
+                    }
+                    ParameterFilePath   = Join-Path '$(parametersRepoRoot)' '${{ deploymentBlock.path }}'
+                    TokenPrefix         = $Settings.parameterFileTokens.tokenPrefix
+                    TokenSuffix         = $Settings.parameterFileTokens.tokenSuffix
+                }
+
+                if ($Settings.parameterFileTokens.localTokens) {
+                  $tokenMap = @{}
+                  foreach ($token in $Settings.parameterFileTokens.localTokens) {
+                    $tokenMap += @{ $token.name = $token.value }
+                  }
+                  Write-Verbose ('Using local tokens [{0}]' -f ($tokenMap.Keys -join ', ')) -Verbose
+                  $ConvertTokensInputs.Tokens += $tokenMap
+                }
+
+                if(-not [String]::IsNullOrEmpty('${{ deploymentBlock.customParameterFileTokens }}')) {
+                  $customTokens = '${{ deploymentBlock.customParameterFileTokens }}' | ConvertFrom-Json -AsHashTable
```

Comment:
In theory I agree (even though this is now a YAML property, not PowerShell). But this would mean we should also make ALL other YAML parameters Pascal Case and this will render this review a nightmare. Let's do this in a separate PR if need be